dev/com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.impl/src/com/ibm/ws/microprofile/reactive/messaging/kafka/adapter/impl/ProducerRecordFactory.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.impl;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import com.ibm.ws.microprofile.reactive.messaging.kafka.adapter.BetaUtils;
+
+public class ProducerRecordFactory {
+
+    private static final String CLAZZ = ProducerRecordFactory.class.getName();
+    private static final Logger LOGGER = Logger.getLogger(CLAZZ);
+
+    public static <K, V> org.apache.kafka.clients.producer.ProducerRecord<K, V> newDelegateProducerRecord(String configuredTopic, String channelName, V value) {
```

Comment:
Technically, the type variables don't match up here.

You can pass in a `ProducerRecord<String, String>` as the `value` (i.e. `V` = `ProducerRecord<String, String>`) and you'll get back a `ProducerRecord<String, String>` (i.e. `V` = `String`).

It might be clearer to make `value` an `Object` to acknowledge that there's no type safety here. It might even make sense to return `ProducerRecord<?, ?>` if we don't actually need to know K and V.

Reply:
yeah, also a good point